server/src/main/java/org/opensearch/common/util/concurrent/QueueResizableOpenSearchThreadPoolExecutor.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util.concurrent;
+
+import org.opensearch.common.ExponentiallyWeightedMovingAverage;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+/**
+ * An extension to thread pool executor, which allows to adjusts the queue size of the
+ * {@code ResizableBlockingQueue} and tracks EWMA.
+ *
+ * @opensearch.internal
+ */
+public final class QueueResizableOpenSearchThreadPoolExecutor extends OpenSearchThreadPoolExecutor
+    implements
+        EWMATrackingThreadPoolExecutor {
+
+    // This is a random starting point alpha. TODO: revisit this with actual testing and/or make it configurable
+    public static double EWMA_ALPHA = 0.3;
```

Comment:
Conceptually this ties "a resizable thread pool executor" to EWMA. I would name everything without EWMA I think. Don't feel too strongly about this though.

Reply:
:+1: moved to `EWMATrackingThreadPoolExecutor`